recipe/recipe.yaml

```diff
@@ -0,0 +1,152 @@
+        import os
+        import platform
+
+        prefix = Path(os.environ["PREFIX"])
+
+        ncurses_libraries = [
+            "libncurses",
+            "libtinfo",
+            "libform",
+            "libmenu",
+            "libpanel",
+        ]
+
+        for each_ncurses_library in ncurses_libraries:
+            assert not (prefix / "lib" / f"{each_ncurses_library}.a").exists()
+            assert not (prefix / "lib" / f"{each_ncurses_library}w.a").exists()
+        if platform.system() == "Darwin":
+            assert (prefix / "lib" / f"{each_ncurses_library}.dylib").exists()
+            assert (prefix / "lib" / f"{each_ncurses_library}w.dylib").exists()
+        if platform.system() == "Linux":
+            assert (prefix / "lib" / f"{each_ncurses_library}.so").exists()
+            assert (prefix / "lib" / f"{each_ncurses_library}w.so").exists()
+
+        # Test include directories
+        assert (prefix / "include" / "ncurses").exists()
+        assert (prefix / "include" / "ncursesw").exists()
+
+        ncurses_headers = [
+            "curses.h",
+            "cursesapp.h",
+            "cursesf.h",
+            "cursesm.h",
+            "cursesp.h",
+            "cursesw.h",
+            "cursslk.h",
+            "eti.h",
+            "etip.h",
+            "form.h",
+            "menu.h",
+            "ncurses.h",
+            "ncurses_dll.h",
+            "panel.h",
+            "term.h",
+            "term_entry.h",
+            "termcap.h",
+            "unctrl.h",
+        ]
+
+        for each_ncurses_header in ncurses_headers:
+            assert (prefix / "include" / "ncurses" / each_ncurses_header).exists()
+            assert (prefix / "include" / "ncursesw" / each_ncurses_header).exists()
+            assert (prefix / "include" / each_ncurses_header).exists()
+
+        ncurses_pc_files = ["form", "menu", "ncurses++", "ncurses", "panel", "tinfo"]
+
+        for each_ncurses_pc_file in ncurses_pc_files:
+            assert (prefix / "lib" / "pkgconfig" / f"{each_ncurses_pc_file}.pc").exists()
+            print((prefix / "lib" / "pkgconfig" / f"{each_ncurses_pc_file}.pc").read_text())
+            assert (prefix / "lib" / "pkgconfig" / f"{each_ncurses_pc_file}w.pc").exists()
+            print((prefix / "lib" / "pkgconfig" / f"{each_ncurses_pc_file}w.pc").read_text())
+
+    requirements:
+      build:
+        - python
+
+  - script:
+      # Test ncurses library arguments.
+      - pkg-config ncurses --libs
+      - pkg-config ncurses --libs | grep "^-L${PREFIX}/lib .*-lncurses -ltinfo$"
+      - pkg-config tinfo --libs | grep "^-L${PREFIX}/lib .*-ltinfo$"
+      - pkg-config ncursesw --libs | grep "^-L${PREFIX}/lib .*-lncursesw -ltinfow$"
+      - pkg-config tinfow --libs | grep "^-L${PREFIX}/lib .*-ltinfow$"
+    requirements:
+      run:
+        - pkg-config
+
+  - script:
+      interpreter: python
+      file: python_ncurses_test.py
+    requirements:
+      run:
+        - python
```

Comment:
This adds a dependency (even though it's at runtime), so let's not do this. Better option is to use `test_downstreams` equivalent in rattler-build.

(Note that the non-rattler-build test was wrong as it was testing the ncurses functionality of the python that conda-build was running on.)

Reply:
Why is it better? From rattler-build perspective it doesn't really matter. In the test we're explicitly importing `ncurses` and I am not sure if Python downstream tests would fail the same way.

Reply:
From any bootstrapping perspective including migrations, it is better to not have cycles.

> In the test we're explicitly importing ncurses and I am not sure if Python downstream tests would fail the same way.

We should add those tests to python if they don't exist.